Dumper/TranslationDumper.php

```diff
@@ -207,14 +207,33 @@ private function dumpTranslationsPerDomain($pattern, array $formats, $target)
         }
     }
 
-    private function dumpTranslationsPerLocale($pattern, array $formats, $target)
+    private function dumpTranslationsPerLocale($pattern, array $formats, $target, $mergeFallback)
     {
-        foreach ($this->getTranslations() as $locale => $domains) {
+        $allTranslations = $this->getTranslations();
+
+        foreach ($allTranslations as $locale => $domains) {
+            $translations = array($locale => $domains);
+
+            if ($mergeFallback) {
+                $localesFallback = array_unique(array(current(explode('_', $locale)), $this->localeFallback));
+                foreach ($localesFallback as $localeFallback) {
+                    if (!empty($localeFallback) && $locale !== $localeFallback && isset($allTranslations[$localeFallback])) {
+                        foreach ($allTranslations[$localeFallback] as $domain => $messages) {
+                            foreach ($messages as $key => $message) {
+                                if (!isset($translations[$locale][$domain][$key])) {
+                                    $translations[$localeFallback][$domain][$key] = $message;
```

Comment:
if you have multiple fallback locales, the merging should be smarter:

- first fallback locale should load all keys not present in the locale (already handled here)
- further fallback locales should load only keys not present in the locale *and any previous fallback locales*

this can be handles by building a second array in parallel, and using that one for detection:

```php
$translations = array($locale => $domains);
$knowTranslations = $domains;

// Asuming proper building of the list of fallbacks
foreach ($localesFallback as $localeFallback) {
    if (!isset($allTranslations[$localeFallback])) {
        continue;
    }

    foreach ($allTranslations[$localeFallback] as $domain => $messages) {
        foreach ($messages as $key => $message) {
            if (!isset($knowTranslations[$domain][$key])) {
                $translations[$localeFallback][$domain][$key] = $message;
                $knowTranslations[$domain][$key] = $message;
            }
        }
    }
}